gedi/process_l2a.py

```diff
@@ -0,0 +1,290 @@
+    canopy_p10 = _compute_nan_percentile(canopy_returns, 10)
+    canopy_p25 = _compute_nan_percentile(canopy_returns, 25)
+    canopy_p50 = _compute_nan_percentile(canopy_returns, 50)
+    canopy_p75 = _compute_nan_percentile(canopy_returns, 75)
+    canopy_p90 = _compute_nan_percentile(canopy_returns, 90)
+
+    d01 = np.ma.masked_outside(rh, canopy_threshold, 5).count(
+        axis=1
+    )  # % returns >= canopy, <=5m
+    d02 = np.ma.masked_outside(rh, 5, 10).count(axis=1)  # % returns >= 5m, <=10m
+    d03 = np.ma.masked_outside(rh, 10, 20).count(axis=1)  # % returns >= 10m, <=20m
+    d04 = np.ma.masked_outside(rh, 20, 30).count(axis=1)  # % returns >= 20m, <=30m
+
+    df["canopy_max"] = pd.Series(canopy_max)
+    df["canopy_min"] = pd.Series(canopy_min)
+    df["canopy_std"] = pd.Series(canopy_std)
+    df["canopy_avg"] = pd.Series(canopy_avg)
+    df["canopy_p10"] = pd.Series(canopy_p10)
+    df["canopy_p25"] = pd.Series(canopy_p25)
+    df["canopy_p50"] = pd.Series(canopy_p50)
+    df["canopy_p75"] = pd.Series(canopy_p75)
+    df["canopy_p90"] = pd.Series(canopy_p90)
+    df["dns"] = pd.Series(dns)
+    df["d01"] = pd.Series(d01)
+    df["d02"] = pd.Series(d02)
+    df["d03"] = pd.Series(d03)
+    df["d04"] = pd.Series(d04)
+
+
+def df_to_geojson(df, outfile):
+    """
+    Convert pandas dataframe to GeoJSON file and save as given file name and path.
+    This file type can be useful when working locally in GIS software.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The DataFrame, assumed to be the output of gedi_L2A_to_df, with or without
+        canopy height metrics appended.
+    outfile : str
+        The location and filename of output json file.
+
+    Returns
+    -------
+    None
+    """
+    df["geometry"] = df.apply(
+        lambda row: Point(row.lon_lowestmode, row.lat_lowestmode), axis=1
+    )
+    GeoDF = gp.GeoDataFrame(df)
+    GeoDF = GeoDF.drop(columns=["lat_lowestmode", "lon_lowestmode"])
+    GeoDF.to_file(outfile, driver="GeoJSON")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-d",
+        "--dir",
+        help=(
+            "The directory containing hd5 files, formatted with a trailing slash,"
+            " such that {dir}{fname} is a valid path, for fname a valid file name."
+        ),
+    )
+    parser.add_argument(
+        "-o",
+        "--outfile",
+        help=(
+            "The stem of the name of the output file, without file extension, "
+            "optional. Default value is 'gedi_output'."
+        ),
+        default="gedi_output",
+    )
+    parser.add_argument(
+        "-f",
+        "--filetype",
+        help="The type of file to output. Acceptable formats are: csv, parquet, GeoJSON.",
+        default="parquet",
+    )
+    args = parser.parse_args()
+
+    ul_lat = 43.38
+    ul_lon = -124.1
+    lr_lat = 42.88
+    lr_lon = -123.6
+    bbox = [ul_lat, ul_lon, lr_lat, lr_lon]
```

Comment:
A decent case could be made that a list-valued command line argument to define the bounding box should be added, since this is the only place in the whole file that would need to get actively changed when considering a different bounding box.